wit/tls.wit

```diff
@@ -0,0 +1,309 @@
+interface tls {
+    use wasi:io/streams@0.2.0.{input-stream, output-stream};
+    use wasi:io/poll@0.2.0.{pollable};
+
+    /// TLS protocol version.
+    ///
+    /// At the time of writing, these are the existing TLS versions:
+    /// - 0x0200: SSLv2 (Deprecated)
+    /// - 0x0300: SSLv3 (Deprecated)
+    /// - 0x0301: TLSv1.0 (Deprecated)
+    /// - 0x0302: TLSv1.1 (Deprecated)
+    /// - 0x0303: TLSv1.2
+    /// - 0x0304: TLSv1.3
+    ///
+    /// TODO: Want to use regular WIT `enum` type, but then adding a new protocol is backwards incompatible.
+    type protocol-version = u16;
```

Comment:
I think it depends on the runtime, for example, let's say a runtime tries to pass TLS 1.4 case to a component export, where the component only supports TLS 1.3 and lower. I'd expect such a call to be disallowed in the runtime, so the function would simply not run.

Similarly, if the runtime would need to return an unsupported version of an enum back to the component via it's import, I'd guess the runtime would have a choice to either handle that case gracefully in some application-specific way (e.g. having an `other(string)` case in the enum) or simply causing a trap in the component.

Perhaps, if we're talking about a "list of requested TLS versions", the runtime could simply omit the unsupported cases, so if a client requested TLS 1.3 *or* TLS 1.4, the runtime could only give the component a choice of TLS 1.3, since TLS 1.4 is not supported by the component's interface.

Reply:
> - I'd expect such a call to be disallowed in the runtime, so the function would simply not run.
> - I'd guess the runtime would have a choice to either handling that (...) causing a trap in the component.

I don't want the guest's security to be limited by the interface version they're targeting. I want the underlying TLS implementation/protocols/cipher-suites/etc to be able to evolve faster than & independently from the API that the the application uses to talk to the implementation.